tests/test_lit_server.py

```diff
@@ -119,3 +121,14 @@ def test_run():
     assert '{"output":16.0}' in output
     os.remove("client.py")
     process.kill()
+
+
+@pytest.mark.asyncio()
+async def test_stream(simple_stream_api):
```

Comment:
We need to add tests for when:
- predict fails during streamed generation
- a timeout happens during streaming

Reply:
Regarding timeout, there are few caveats @lantiga :
* FastAPI seem to send the streaming content to BackgroundTask so we can't raise TimeoutException during streaming. 
* `server = LitServer(simple_stream_api, stream=True, timeout=10)` -> client will be able to read whatever server could produce within those 10 seconds of timeout.

Reply:
opened the issue [here](https://github.com/Lightning-AI/litserve/issues/46)